### PR TITLE
Shell scripts: More Shellcheck fixes

### PR DESCRIPTION
**Pull request by bje-**
## What does this PR do?

Sillences more warnings from Shellcheck. Please ask if it is not clear *why* any of them are needed.

### CODE REVIEW

File: gdal/scripts/detect_tabulations.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ cd "$GDAL_ROOT"
 ret_code=0
 
 echo "Checking for tabulation characters..."
-if find alg gnm port ogr gcore frmts apps \( -name "*.cpp" -o -name "*.h" \) -exec sh -c "grep -P '\\t' {} >/dev/null && echo {}" \; | grep -v frmts/msg/PublicDecompWT | grep -v frmts/jpeg/libjpeg | grep -v frmts/gtiff/libtiff | grep -v frmts/gtiff/libgeotiff | grep -v frmts/grib/degrib | grep -v ogr/ogrsf_frmts/geojson/libjson | grep -v frmts/hdf4/hdf-eos | grep -v frmts/gif/giflib | grep -v frmts/pcraster/libcsf | grep -v frmts/png/libpng | grep -v cpl_mem_cache ; then
+if grep -r --files-with-matches -P --include='alg/*.h alg/*.cpp gnm/*.h gnm/*.cpp port/*.h port/*.cpp ogr/*.h ogr/*.cpp gcore/*.h gcore/*.cpp frmts/*.h frmts/*.cpp apps/*.h apps/*.cpp' '\\t' | grep -v -e frmts/msg/PublicDecompWT -e frmts/jpeg/libjpeg -e frmts/gtiff/libtiff -e frmts/gtiff/libgeotiff -e frmts/grib/degrib -e ogr/ogrsf_frmts/geojson/libjson -e frmts/hdf4/hdf-eos -e frmts/gif/giflib -e frmts/pcraster/libcsf -e frmts/png/libpng -e cpl_mem_cache ; then
```
Comment (rouault) [inline gdal/scripts/detect_tabulations.sh]: No idea what's wrong in that change, but in practice it doesn't scan any file...

File: gdal/scripts/fix_typos.sh [hunk 2]
```diff
@@ -48,18 +48,18 @@ cd "$GDAL_ROOT"
 if ! test -d fix_typos; then
     # Get our fork of codespell that adds --words-white-list and full filename support for -S option
     mkdir fix_typos
-    cd fix_typos
-    git clone https://github.com/rouault/codespell
-    git checkout codespell/gdal_improvements
```
Comment (rouault) [inline gdal/scripts/fix_typos.sh]: This was actually broken from one of your previous changes.

```git checkout codespell/gdal_improvements``` errors out for me with
```error: pathspec 'codespell/gdal_improvements' did not match any file(s) known to git.```
The previous version that cd'ed to codespell and did git checkout gdal_improvements is the only one to work

File: gdal/scripts/setdevenv.sh [hunk 3]
```diff
@@ -14,26 +14,26 @@ if [[ $called == "$0" ]]; then
     exit 1
 fi
 
-GDAL_ROOT=$(cd $(dirname ${BASH_SOURCE[0]})/..; pwd)
+GDAL_ROOT=$(realpath -e $(dirname ${BASH_SOURCE[0]})/..)
```
Comment (rouault) [inline gdal/scripts/setdevenv.sh]: Apparently realpath is not available on Mac: https://github.com/google/enjarify/issues/9

File: gdal/scripts/fix_typos.sh (hunk 2 above)
Comment (bje-) [reply gdal/scripts/fix_typos.sh]: Fixed.

File: gdal/scripts/setdevenv.sh (hunk 3 above)
Comment (bje-) [reply gdal/scripts/setdevenv.sh]: OK, will use `readlink(1)` instead.

File: gdal/scripts/fix_typos.sh [hunk 4]
```diff
@@ -50,7 +50,7 @@ if ! test -d fix_typos; then
     mkdir fix_typos
     (cd fix_typos
      git clone https://github.com/rouault/codespell
-     git checkout codespell/gdal_improvements
+     (cd codespell && git checkout gdal_improvements)
```
Comment (rouault) [inline gdal/scripts/fix_typos.sh]: all the code below must also occur in the codespell subdirectory

File: gdal/scripts/setdevenv.sh [hunk 5]
```diff
@@ -14,7 +14,7 @@ if [[ $called == "$0" ]]; then
     exit 1
 fi
 
-GDAL_ROOT=$(realpath -e $(dirname ${BASH_SOURCE[0]})/..)
+GDAL_ROOT=$(readlink -e $(dirname ${BASH_SOURCE[0]})/..)
```
Comment (rouault) [inline gdal/scripts/setdevenv.sh]: are you sure that works on Mac ?

File: gdal/scripts/fix_typos.sh (hunk 4 above)
Comment (bje-) [reply gdal/scripts/fix_typos.sh]: Really? The very original version did this:

```
cd codespell
git checkout gdal_improvements
cd ..
```

File: gdal/scripts/setdevenv.sh (hunk 5 above)
Comment (bje-) [reply gdal/scripts/setdevenv.sh]: Apparently it will, but not with `-e`. :-(

File: gdal/scripts/fix_typos.sh (hunk 4 above)
Comment (rouault) [reply gdal/scripts/fix_typos.sh]: You're right. Forget my above comment

File: gdal/scripts/detect_tabulations.sh [hunk 6]
```diff
@@ -19,7 +19,7 @@ cd "$GDAL_ROOT"
 ret_code=0
 
 echo "Checking for tabulation characters..."
-if find alg gnm port ogr gcore frmts apps \( -name "*.cpp" -o -name "*.h" \) -exec sh -c "grep -P '\\t' {} >/dev/null && echo {}" \; | grep -v frmts/msg/PublicDecompWT | grep -v frmts/jpeg/libjpeg | grep -v frmts/gtiff/libtiff | grep -v frmts/gtiff/libgeotiff | grep -v frmts/grib/degrib | grep -v ogr/ogrsf_frmts/geojson/libjson | grep -v frmts/hdf4/hdf-eos | grep -v frmts/gif/giflib | grep -v frmts/pcraster/libcsf | grep -v frmts/png/libpng | grep -v cpl_mem_cache ; then
+if grep -r --files-with-matches -P --include='*.h' --include='*.cpp' '\\t' alg gnm port ogr gcore frmts apps | grep -v -e frmts/msg/PublicDecompWT -e frmts/jpeg/libjpeg -e frmts/gtiff/libtiff -e frmts/gtiff/libgeotiff -e frmts/grib/degrib -e ogr/ogrsf_frmts/geojson/libjson -e frmts/hdf4/hdf-eos -e frmts/gif/giflib -e frmts/pcraster/libcsf -e frmts/png/libpng -e cpl_mem_cache ; then
```
Comment (rouault) [inline gdal/scripts/detect_tabulations.sh]: something wrong. This wrongly detects all source files as having tabulations

Comment (bje-) [reply gdal/scripts/detect_tabulations.sh]: I worked through it by hand. The problem is that the tab was overly escaped. Fixed now.

**Pull request merged**